### PR TITLE
feat(service-registry-api): add service-registry-api package

### PR DESCRIPTION
**Pull request by maancham**
## Description

AXE-4789

## Todos

- [ ] Unit tests
- [ ] Manual tests
- [ ] Documentation
- [ ] Connect epics/issues

### CODE REVIEW

**Comment by cjcobb23** [comment]
why do we need this?

**Comment by cgorenflo** [comment]
> why do we need this?

to break the cyclic dependency between service-registry and coordinator

**Comment by cjcobb23** [comment]
> > why do we need this?
> 
> to break the cyclic dependency between service-registry and coordinator

For what exactly? Is there a planned change that will create a cyclic dependency?

File: packages/service-registry-api/src/primitives.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                Self {
+                    bonding_state: BondingState::Unbonded,
+                    ..self
+                },
+                amount,
+            )),
+            _ => Err(ContractError::InvalidBondingState(self.bonding_state)),
+        }
+    }
+}
```
Comment (cgorenflo) [inline packages/service-registry-api/src/primitives.rs:102]: we'll have to look into how the verifier is used in other contracts. It doesn't make sense to provide this logic outside of the service-registry

File: packages/service-registry-api/src/primitives.rs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+
+#[cw_serde]
+pub struct WeightedVerifier {
+    pub verifier_info: Verifier,
+    pub weight: nonempty::Uint128,
+}
+
+/// For now, all verifiers have equal weight, regardless of amount bonded
+pub const VERIFIER_WEIGHT: nonempty::Uint128 = nonempty::Uint128::one();
```
Comment (cgorenflo) [inline packages/service-registry-api/src/primitives.rs:111]: state definitely shouldn't go into this crate

File: packages/service-registry-api/src/primitives.rs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    NotAuthorized,
+    Authorized,
+    Jailed,
+}
+
+type ServiceName = String;
+type VerifierAddress = Addr;
+
+pub const SERVICES: Map<&ServiceName, Service> = Map::new("services");
+pub const VERIFIERS: Map<(&ServiceName, &VerifierAddress), Verifier> = Map::new("verifiers");
```
Comment (cgorenflo) [inline packages/service-registry-api/src/primitives.rs:148]: no state

**Review by cgorenflo — CHANGES_REQUESTED** [review]

**Review by cgorenflo — APPROVED** [review]

**Pull request merged**